addons/account/models/account_payment.py

```diff
@@ -101,71 +98,138 @@ class account_register_payments(models.TransientModel):
+        active_ids = self._context.get('active_ids')
+
+        # Check for selected invoices ids
+        if not active_ids:
+            raise UserError(_("Programmation error: wizard action executed without active_ids in context."))
+
+        invoice_ids = self.env['account.invoice'].browse(active_ids)
+
+        # Check all invoices are open
+        if any(invoice.state != 'open' for invoice in invoice_ids):
             raise UserError(_("You can only register payments for open invoices"))
-        if any(inv.commercial_partner_id != invoices[0].commercial_partner_id for inv in invoices):
-            raise UserError(_("In order to pay multiple invoices at once, they must belong to the same commercial partner."))
-        if any(MAP_INVOICE_TYPE_PARTNER_TYPE[inv.type] != MAP_INVOICE_TYPE_PARTNER_TYPE[invoices[0].type] for inv in invoices):
-            raise UserError(_("You cannot mix customer invoices and vendor bills in a single payment."))
-        if any(inv.currency_id != invoices[0].currency_id for inv in invoices):
+        # Check all invoices have the same currency
+        if any(inv.currency_id != invoice_ids[0].currency_id for inv in invoice_ids):
             raise UserError(_("In order to pay multiple invoices at once, they must use the same currency."))
 
-        total_amount = sum(inv.residual * MAP_INVOICE_TYPE_PAYMENT_SIGN[inv.type] for inv in invoices)
-        communication = ' '.join([ref for ref in invoices.mapped('reference') if ref])
+        # Look if we are mixin multiple commercial_partner or customer invoices with vendor bills
+        multi = any(inv.commercial_partner_id != invoice_ids[0].commercial_partner_id
+            or MAP_INVOICE_TYPE_PARTNER_TYPE[inv.type] != MAP_INVOICE_TYPE_PARTNER_TYPE[invoice_ids[0].type]
+            for inv in invoice_ids)
+
+        total_amount = self._compute_payment_amount(invoice_ids)
 
         rec.update({
             'amount': abs(total_amount),
-            'currency_id': invoices[0].currency_id.id,
+            'currency_id': invoice_ids[0].currency_id.id,
             'payment_type': total_amount > 0 and 'inbound' or 'outbound',
-            'partner_id': invoices[0].commercial_partner_id.id,
-            'partner_type': MAP_INVOICE_TYPE_PARTNER_TYPE[invoices[0].type],
-            'communication': communication,
+            'partner_id': False if multi else invoice_ids[0].commercial_partner_id.id,
+            'partner_type': False if multi else MAP_INVOICE_TYPE_PARTNER_TYPE[invoice_ids[0].type],
+            'communication': ' '.join([ref for ref in invoice_ids.mapped('reference') if ref]),
+            'invoice_ids': [(6, 0, invoice_ids.ids)],
+            'multi': multi,
         })
         return rec
 
+    @api.model
     def get_payment_vals(self):
-        """ Hook for extension """
+        '''Return the a values dictionary to create a new payment object.
+
+        :return: Payment values as a dict.
+        '''
+        if self.multi:
```

Comment:
If we are not in multi (it means not dealing with different partners/ invoice type), the values are already computed on the wizard. If multi, we are dealing with a subset of invoices passed through the context.